### ISSUE TITLE
Ambiguous reference to member 'withClass', with three "candidate" copies of TyphoonDefinition

### ISSUE THREAD

**Issue by jraqula** [appsquickly/Typhoon#484]
I am working on an iOS app, and include Typhoon using either CocoaPods or source-as-subproject. In either case, I get this strange error. Typhoon links & imports correctly, however the Swift compiler complains that my use of `withClass` is ambiguous, and refers to three different `Typhoon.TyphoonDefinition`s for the source. However, it doesn't provide any info about why there are 3, or how they are different.

See my StackOverflow question for images: http://stackoverflow.com/questions/36120533/typhoon-di-framework-produces-ambiguous-reference-error

**Comment by alexgarbarev** [comment]
Hmm.. Did you solve that problem? If not, can you share demo project (so anyone can easily face with that issue)?

**Comment by jraqula** [comment]
hi there, thanks for responding. I solved the issue for myself and made note on SO, but forgot to comment here.

I first installed Typhoon as a dependency by Carthage, but decided to undo that bc of some compatibility issues I was having with other dependencies. So i reinstalled it as a submodule. My best guess is there were some artifacts from this that left behind, and Xcode found duplicate symbols there or something -- once I cleaned out build folder and all derived data folders, etc, then rebooted everything, it compiled OK and the issue is gone.
